### DISCUSSION TITLE
[API Request] Allow sort order for getAlbumList2

### DISCUSSION THREAD

**Discussion by jeffvli** [opensubsonic/open-subsonic-api#19] [API requests]
### Type of change

API tweak

### Proposal description

Would also be used for other list endpoints such as #18.

There's currently no way to change the sort order for `getAlbumList2` besides using `fromYear` and `toYear` to reverse the chronological sort.

An example grievance would be using `type=alphabeticalByName` and then having to paginate through your entire library to reach an album that starts with the letter `Z`. This becomes more difficult with larger libraries.

This proposal would add an optional parameter `order` to set descending or ascending sort for the resulting query.

### Backward compatibility impact

If `order` is omitted, the sort order will use the original order set by the `type` parameter.

### Backward compatibility

- [X] No backward compatibility impact.

### API details

```
Parameters:
- order -- accepts a string of either "desc" or "asc"

GET /getAlbumList2?type=alphabeticalByName&order=desc

{
  "subsonic-response": {
    "status": "ok",
    "version": "1.16.1",
    "type": "AwesomeServerName",
    "serverVersion": "0.1.3 (tag)",
    "albumList2": {
      "album": [
        {
          "title": "Z - Forget and Remember",
          "name": "Z - Forget and Remember",
        },
        {
          "title": "X - Buried in Nausea",
          "name": "X - Buried in Nausea",
        }
      ]
    }
  }
}

```

### Alternative solutions

It can be argued that if a user wanted to find something specific, they could just use the `search3` endpoint to find what they're looking for instead of needing to reverse the sort order. But I think having a way to set the sort order on the list would greatly improve the user experience if a user did not know exactly what they were looking for.

**Comment by dweymouth** [comment]
Related to #7
